### ISSUE TITLE
TypeVarTuple example from python docs yields error

### ISSUE THREAD

**Issue by toolslive** [microsoft/pyright#6906]
**Describe the bug**
[this example](https://docs.python.org/3/library/typing.html#typing.TypeVarTuple) from the python website is problematic:
```python
T = TypeVar("T")
Ts = TypeVarTuple("Ts")

def move_first_element_to_last(tup: tuple[T, *Ts]) -> tuple[*Ts, T]:
    return (*tup[1:], tup[0])

```

![lovely](https://github.com/microsoft/pyright/assets/1249832/e62eedd8-eb93-4f08-995c-f1332c1af8e8)
via the cli:
```
geweldig.py:7:12 - error:  Expression of type "tuple[*tuple[T@move_first_element_to_last | Union[*Ts@move_first_element_to_last], ...], T@move_first_element_to_last]" cannot be assigned to return type "tuple[*Ts@move_first_element_to_last, T@move_first_element_to_last]"
    "tuple[*tuple[T@move_first_element_to_last | Union[*Ts@move_first_element_to_last], ...], T@move_first_element_to_last]" is incompatible with "tuple[*Ts@move_first_element_to_last, T@move_first_element_to_last]"
      Tuple entry 1 is incorrect type
        Type "*tuple[T@move_first_element_to_last | Union[*Ts@move_first_element_to_last], ...]" cannot be assigned to type "*Ts@move_first_element_to_last" (reportGeneralTypeIssues)
```

```
$> pyright --version
pyright 1.1.344
...

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release of pyright.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.345, which I just published. It will also be included in a future release of pylance.
